### ISSUE TITLE
Option string values parsed as numbers

### ISSUE THREAD

**Issue by jvolkman** [emicklei/proto#129]
Hi. I'm using this as part of https://github.com/stackb/rules_proto and I'm running into some parsing problems with proto files that contain custom option values. It seems that string values are being parsed as numeric literals.

Option values below aren't valid numeric representations, but it shouldn't matter because they're strings.

`test.proto`
```protobuf
syntax = "proto3";

import "google/protobuf/descriptor.proto";

extend google.protobuf.EnumValueOptions {
  string enum_opt = 99999999;
}

enum EnumTest {
  VAL0 = 0 [(enum_opt) = '09'];
  VAL1 = 1 [(enum_opt) = '8P'];
  VAL2 = 2 [(enum_opt) = '7E'];
}
```

`test.go`
```go
package main

import (
	"fmt"
	"os"

	"github.com/emicklei/proto"
)

func main() {
	reader, _ := os.Open("test.proto")
	defer reader.Close()

	parser := proto.NewParser(reader)
	_, err := parser.Parse()
	if err != nil {
		fmt.Println(err)
	}
}
```

```shell
$ go run /tmp/test.go
go scanner error at <input>:10:27 = invalid digit '9' in octal literal
go scanner error at <input>:11:27 = 'P' exponent requires hexadecimal mantissa
go scanner error at <input>:11:27 = exponent has no digits
go scanner error at <input>:12:27 = exponent has no digits
```

**Comment by emicklei** [comment]
thank you for reporting this and providing an example. I will look at it in more detail

**Comment by emicklei** [comment]
this one passes

```
func TestStringValuesParsedAsNumbers(t *testing.T) {
	src := `VAL0 = 0 [(enum_opt) = '09'];`
	f := new(EnumField)
	if err := f.parse(newParserOn(src)); err != nil {
		t.Fatal(err)
	}
	if got, want := f.ValueOption.Constant.Source, "09"; got != want {
		t.Errorf("got [%v:%T] want [%v:%T]", got, got, want, want)
	}
}
```

**Comment by emicklei** [comment]
but as a Program it fails... hmmm

**Comment by emicklei** [comment]
have a look at https://github.com/emicklei/proto/pull/130

**Comment by jvolkman** [comment]
Thanks. So it looks like another workaround would be to just use double quotes for these values? I guess handling of double quoted strings is built into text/scanner?

**Comment by emicklei** [comment]
yes that is correct. for a long time (and now extended) the package was able to use the Go scanner for getting all the tokens.  This choice prevented me from creating my own dedicated scanner that would not need those workarounds. Option syntax has always been very tricky to parse also because their is no clear definition for it. Look at the other option tests to see what others stumbled upon.

**Comment by jvolkman** [comment]
Got it. Yeah I wish Google would release some more formal specs on proto options. The short forms are pretty simple, but the long form is embedded protobuf text format for which there's not really a public definition (they have it described internally, though). This grammar is pretty complete: https://github.com/jvolkman/intellij-protobuf-editor/blob/master/core/src/main/bnf/textformat.bnf

**Comment by jvolkman** [comment]
Looks like this is resolved in v1.9.2. Thanks @emicklei!
